insert_coverage_stats.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+"""Generate summary statistics of read coverage by region defined in a bedfile.
+Use to evaluate per-amplicon performance of primers for a sample."""
+
+import argparse
+from io import StringIO
+import subprocess
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("bedfile", help="Amplicon insert regions bedfile path")
+    parser.add_argument("alignments", help="Alignments .bam path")
+    parser.add_argument(
+        "--prop_cov",
+        help="Proportion of a read that must be mapped to a region "
+        "for it to count towards coverage. Default: 0.90",
+        default="0.90",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def run(args):
+    # Run bedtools to get per-base coverage for reads that map to each insert
+    cmd = [
+        "bedtools",
+        "coverage",
+        "-F",
+        args.prop_cov,
+        "-d",
+        "-a",
+        args.bedfile,
+        "-b",
+        args.alignments,
+    ]
+    a = subprocess.Popen(cmd, stdout=subprocess.PIPE)
```

Comment:
Add check for successful return code here.

Reply:
Done + made this a little more readable by calling `subprocess.run()` instead.